### PR TITLE
Add allele methods to function registry and functions.py

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by cseed** [comment]
Hmm.  I thought we were going to get rid of AltAllele and instead have `alleles: Array[String]` and e.g. `is_snp(alleles, 1)`.

**Comment by tpoterba** [comment]
Isn't that what this is? They just take strings.

**Comment by tpoterba** [comment]
I don't like taking an array and an int because it makes the general case difficult, like if I've got two cols of a tsv

**Comment by cseed** [comment]
(actually looking at the code...) Oh, I see, they take pair of strings.  Yeah, I like that even better, nice.

File: src/main/scala/is/hail/variant/AltAllele.scala [hunk 1]
```diff
@@ -41,14 +44,114 @@ object AltAllele {
...
+    val altChar = alt(pos)
+    (refChar == 'A' && altChar == 'G') || (refChar == 'G' && altChar == 'A') ||
+      (refChar == 'C' && altChar == 'T') || (refChar == 'T' && altChar == 'C')
+  }
+
+  def hamming(a1: String, a2: String): Int = {
+    if (a1.length != a2.length)
+      fatal(s"function 'hamming' requires strings to have equal length, " +
+        s"found lengths ${a1.length} and ${a2.length}: '$a1', '$a2'")
+    require(a1.length == a2.length)
```
Comment (jigold) [inline src/main/scala/is/hail/variant/AltAllele.scala]: delete

**Comment by tpoterba** [comment]
I agree it's getting pretty big. Let's think about docs after the `from hail2 import *` => `import hl` reorg

**Pull request merged**